exporter/otlpexporter/mock_receiver.go

```diff
@@ -0,0 +1,78 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otlpexporter // import "go.opentelemetry.io/collector/exporter/otlpexporter"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/exporter/exportertest"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/otlpreceiver"
+	"go.opentelemetry.io/collector/receiver/receivertest"
+)
+
+// DataReceiverBase implement basic functions needed by all receivers.
+type DataReceiverBase struct {
+	// Port on which to listen.
+	Port int
+}
+
+type BaseOTLPDataReceiver struct {
```

Comment:
No need to export publicly 
```suggestion
type mockReceiver struct {
```

Reply:
Done, but `mockReceiver` already existed in this package in `otlp_test`, so I named it `mockOtlpReceiver`